tests/js/server/aql/aql-traversal-paths-optimization.js

```diff
@@ -0,0 +1,216 @@
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' RETURN [p.edges, p.vertices]", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' RETURN [p.edges, p]", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' RETURN [p.vertices, p]", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' RETURN [p.edges, p.vertices, p]", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' RETURN p.testi", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' COLLECT pl = LENGTH(p) RETURN pl", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' COLLECT key = v._key INTO g RETURN { key, other: g[*].p.edges }", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' FILTER LENGTH(p) > 0 RETURN v", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' SORT p.edges RETURN v", [ false, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' SORT p.vertices RETURN v", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' SORT p.edges, p.vertices RETURN v", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' SORT LENGTH(p.edges) RETURN v", [ false, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' SORT LENGTH(p.vertices) RETURN v", [ true, false, false ]],
+       
+        // PRUNE
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE v.testi == 'abc' RETURN p", [ true, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE v.testi == 'abc' RETURN p.edges", [ false, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE v.testi == 'abc' RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.vertices[-1].testi == 'abc' RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.vertices[-1].testi == 'abc' RETURN p.edges", [ false, true, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.edges[-1].testi == 'abc' RETURN p.vertices", [ true, false, false ]],
+        [ "FOR v, e, p IN 1..1 OUTBOUND '" + vn + "/test0' GRAPH '" + gn + "' PRUNE p.edges[-1].testi == 'abc' RETURN p.edges", [ false, true, false ]],
+
```

Comment:
I think we should add the following test-cases as well:

`FILTER p.edges[*].weight ALL >= 3 RETURN p.vertices` => `vertices:true, edges:false` (the edges should be optimized into Traversal and do not need to be produced)

`FILTER p.vertices[1].weight >= 3 RETURN p.edges` => `vertices:false, edges:true` (same as above)

`FILTER p.edges[0].weight >= 3 RETURN p.vertices` => `vertices:true, edges:false` (the edges should be optimized into Traversal and do not need to be produced)

`FILTER p.vertices[*].weight ALL >= 3 RETURN p.edges` => `vertices:false, edges:true` (same as above)


If they do not work as suggested  I am happy to have them with `true, true` for now, and we have another possible optimization here.